commands/profile.go

```diff
@@ -143,67 +197,47 @@ func getProfileController(cfgFlagValue string) (profile.Controller, error) {
-		return fmt.Errorf("failed to get profile names due to: %w", err)
-	}
-	newProfile := getNewProfile(profiles)
-	if err = profileController.CreateProfile(newProfile); err != nil {
+func CreateProfile(profileController profile.Controller, newProfile entity.Profile) error {
+	if err := profileController.CreateProfile(newProfile); err != nil {
 		return fmt.Errorf("failed to create profile %v due to: %w", newProfile, err)
 	}
 	return nil
 }
 
-// getNewProfile gets new profile information from user using command line
-func getNewProfile(profileMap map[string]entity.Profile) entity.Profile {
-	var name string
-	for {
-		fmt.Printf("Enter profile's name: ")
-		name = getUserInputAsText(checkInputIsNotEmpty)
-		if _, ok := profileMap[name]; !ok {
-			break
-		}
-		fmt.Println("profile", name, "already exists.")
-	}
-	fmt.Printf("Elasticsearch Endpoint: ")
-	endpoint := getUserInputAsText(checkInputIsNotEmpty)
-	fmt.Printf("Is security plugin enabled? Y/N ")
-	isSecured := getUserInputAsBoolean(checkInputIsNotEmpty)
-	var user, password string
-	if isSecured {
-		fmt.Printf("User Name: ")
-		user = getUserInputAsText(checkInputIsNotEmpty)
-		fmt.Printf("Password: ")
-		password = getUserInputAsMaskedText(checkInputIsNotEmpty)
+func validateProfileName(name string, controller profile.Controller) error {
+	profileMap, err := controller.GetProfilesMap()
+	if err != nil {
+		return err
 	}
-	return entity.Profile{
-		Name:     name,
-		Endpoint: endpoint,
-		UserName: user,
-		Password: password,
+	if _, ok := profileMap[name]; !ok {
+		return nil
 	}
+	return fmt.Errorf("profile %s already exists", name)
 }
 
-func getUserInputAsBoolean(isValid func(input string) bool) bool {
-	response := getUserInputAsText(isValid)
-	switch strings.ToLower(response) {
-	case "y", "yes":
-		return true
-	case "n", "no":
-		return false
-	default:
-		fmt.Printf("please type (y)es or (n)o: ")
-		return getUserInputAsBoolean(isValid)
+// getBasicAuthDetails gets new basic HTTP Auth profile information from user using command line
+func getBasicAuthDetails(newProfile *entity.Profile) {
+	fmt.Printf("User Name: ")
+	newProfile.UserName = getUserInputAsText(checkInputIsNotEmpty)
+	fmt.Printf("Password: ")
+	newProfile.Password = getUserInputAsMaskedText(checkInputIsNotEmpty)
+}
+
+// getAWSIAMAuthDetails gets new AWS IAM Auth profile information from user using command line
+func getAWSIAMAuthDetails(newProfile *entity.Profile) {
+	fmt.Printf("AWS profile name (Leave blank if you want to provide credentials using environment variables): ")
+	awsIAM := &entity.AWSIAM{
+		ProfileName: getUserInputAsText(nil),
 	}
+	newProfile.AWS = awsIAM
 }
 
 // getUserInputAsText get value from user as text
 func getUserInputAsText(isValid func(string) bool) string {
 	var response string
 	//Ignore return value since validation is applied below
 	_, _ = fmt.Scanln(&response)
-	if !isValid(response) {
+	if isValid != nil && !isValid(response) {
 		return getUserInputAsText(isValid)
```

Comment:
isValid is a function, hence, isValid is expected to show error message and return with false and caller will determine whether to call again or exit. In future, want to make this is varargs, so that more than one validation can be applied to the user input. I haven't made it yet since we don't have use case yet.